Source/DafnyCore/AST/DafnyAst.cs

```diff
@@ -58,7 +58,7 @@ public abstract class INode {
           continue;
         }
 
-        foreach (var child in current.Children) {
+        foreach (var child in current.Children.Reverse()) {
```

Comment:
`System.Collections.Generic.LinkedList<T>` is a doubly linked list so that can work as the double-ended queue that you mention, but I don't see how it would give me the traversal order that I want. I'm only adding elements at one location, where the `Push` is. If I change that to an `Enqueue` instead, then I get breadth instead of depth first traversal and that's not what I want.